modules/dataquery/test/TestPlan.md

```diff
@@ -0,0 +1,100 @@
+10. Remove all selected fields from right column.
+11. Remove text from fields filter bar.
+12. Click the `Add all` button.
+13. Assert that: clicking the `Clear` button in the `Selected Fields` column effectively clears all selected fields.
+14. Click the `Add all` button.
+15. Assert that: clicking the `trashbin icon` in the `Selected Fields` column effectively removes the selected field **AND** the active selection in the main table.
+16. Assert that: clicking some selected fields in the main table (grey-ish lines in the table) also remove them from the selected column (toggle interaction).
+
+### Visits
+
+1. Select the `Other parameters` field category in the top dropdown bar.
+2. Assert that: default visits are loaded, fields for this category are loaded.
+3. Assert that: clicking on a field such as `VisitLabel` or `Project` greys the line **AND** adds visit information on the same line.
+4. Assert that: removing visits from `Default Visits` allows to reselect them, either by clicking the right arrow, or by entering corresponding text in the text area.
+5. Make sure the `Sync with selected fields` is **NOT** checked out.
+6. Add all visits from `Default Visits` box.
+7. Click some fields in the list.
+8. Assert that: selected fields should have a visit selection box each.
+9. Assert that: all visits should be displayed in the field lines (same as those ni the `Default Visits` box).
+10. Assert that: all selected fields in the right column have all visits described.
+11. Remove some visits from `Default Visits` box.
+12. Click on other fields.
+13. Assert that: new field visits only reflects what the `Default Visits` box is showing.
+14. Assert that: new selected fields in the right column only have the newly default visits.
+15. Check `Sync with selected field` checkbox.
+16. Assert that: fields that were previously with all visits now are only with the selected visits in `Default Visits`.
+17. Make sure field visits are still updatable independantly (line by line).
+18. Make sure if `Default Visits` are changed, it affects all fields.
+
+
+## Filters selection page
+
+1. Make sure no filter are already selected. The sentence `Currently querying for ALL candidates` should be displayed.
+2. MAke sure the blue notification saying `Note that only candidates which you have permission to access in LORIS are included in results. Number of results may vary from other users running the same query.` is there.
+3. Make sure a preview result is displayed in the to-right hand corner.
+4. Make sure the `Add Condition` button triggers field select modal.
+   1. Add a field as a filter.
+   2. Assert that: the condition now appears in the filter list.
+   3. Assert that: `Add Condition` and `Import from CSV` buttons are now replaced by `Add "and" condition` and `Add "or" condition`.
+   4. Assert that: the top-right corner is updated.
+5. Make sure `Import from CSV` button feature an upload modal.
```

Comment:
I think you need to remove the condition added in step 4 for this to work, otherwise the button won't be there.